Compilation Issues with Latest Version of Reflex

Using UniState Version 1.9.0 and Reflex 14.3.0 currently has the following compile errors

Library\PackageCache\com.bazyleu.unistate@a615c8766ec9\Runtime\Integrations\Reflex\ReflexBuildExtensions.cs(131,59): error CS0535: 'ReflexBuildExtensions.ReflexStateMachineResolver' does not implement interface member 'IResolver.Resolution'

Library\PackageCache\com.bazyleu.unistate@a615c8766ec9\Runtime\Integrations\Reflex\ReflexBuildExtensions.cs(131,59): error CS0535: 'ReflexBuildExtensions.ReflexStateMachineResolver' does not implement interface member 'IResolver.DeclaringContainer'

--- a/Assets/Examples/Infrastructure/Reflex/DiceInstaller.cs
+++ b/Assets/Examples/Infrastructure/Reflex/DiceInstaller.cs
@@ -1,6 +1,7 @@
 ﻿using Examples.States;
 using Reflex.Attributes;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 using UnityEngine;
 
@@ -13,20 +14,23 @@ namespace Examples.Infrastructure.Reflex
 
         public void InstallBindings(ContainerBuilder builder)
         {
-            builder.AddStateMachine(typeof(StateMachine), typeof(IStateMachine));
+            builder.RegisterStateMachine(typeof(StateMachine), typeof(IStateMachine));
 
-            builder.AddState(typeof(LostState));
-            builder.AddState(typeof(RollDiceState));
-            builder.AddState(typeof(StartGameState));
-            builder.AddState(typeof(WinState));
+            builder.RegisterState(typeof(LostState));
+            builder.RegisterState(typeof(RollDiceState));
+            builder.RegisterState(typeof(StartGameState));
+            builder.RegisterState(typeof(WinState));
 
-            builder.AddSingleton(container =>
-            {
-                DiceEntryPoint entryPoint = new(container.Resolve<IStateMachine>());
-                entryPoint.Start();
+            builder.RegisterFactory<DiceEntryPoint>(
+                container =>
+                {
+                    DiceEntryPoint entryPoint = new(container.Resolve<IStateMachine>());
+                    entryPoint.Start();
 
-                return entryPoint;
-            });
+                    return entryPoint;
+                },
+                Lifetime.Singleton,
+                global::Reflex.Enums.Resolution.Lazy);
         }
     }
 }

--- a/Assets/UniState/Runtime/Integrations/Reflex/ReflexBuildExtensions.cs
+++ b/Assets/UniState/Runtime/Integrations/Reflex/ReflexBuildExtensions.cs
@@ -1,59 +1,94 @@
 ﻿#if UNISTATE_REFLEX_SUPPORT
 
 using System;
-using System.Collections.Generic;
 using Reflex.Core;
 using Reflex.Enums;
-using Reflex.Resolvers;
 
 namespace UniState
 {
     public static class ReflexBuildExtensions
     {
+        public static void RegisterStateMachine(
+            this ContainerBuilder builder,
+            Type stateMachineImplementation,
+            Type stateMachineContract)
+        {
+            RegisterStateMachine(builder, stateMachineImplementation, stateMachineContract, Lifetime.Transient);
+        }
+
+        public static void RegisterStateMachine(
+            this ContainerBuilder builder,
+            Type stateMachineImplementation,
+            Type stateMachineContract,
+            Lifetime lifetime)
+        {
+            RegisterStateMachineInternal(builder, stateMachineImplementation, stateMachineContract, lifetime);
+        }
+
+        public static void RegisterState(this ContainerBuilder builder, Type state)
+        {
+            RegisterState(builder, state, Lifetime.Transient);
+        }
+
+        public static void RegisterState(this ContainerBuilder builder, Type state, Lifetime lifetime)
+        {
+            ValidateStateBindingInput(state);
+
+            builder.RegisterType(state, GetStateContracts(state), lifetime, Resolution.Lazy);
+        }
+
+        public static void RegisterState(this ContainerBuilder builder, Type stateImplementation, Type stateContract)
+        {
+            RegisterState(builder, stateImplementation, stateContract, Lifetime.Transient);
+        }
+
+        public static void RegisterState(
+            this ContainerBuilder builder,
+            Type stateImplementation,
+            Type stateContract,
+            Lifetime lifetime)
+        {
+            ValidateStateBindingInput(stateImplementation, stateContract);
+
+            builder.RegisterType(
+                stateImplementation,
+                new[] { stateContract },
+                lifetime,
+                Resolution.Lazy);
+        }
+
+        [Obsolete("Use RegisterStateMachine(builder, implementation, contract) or the overload with Lifetime.")]
         public static void AddStateMachine(
             this ContainerBuilder builder,
             Type stateMachineImplementation,
-            Type stateMachineContract)
-        {
-            AddStateMachineInternal(builder, stateMachineImplementation, stateMachineContract, Lifetime.Transient);
-        }
+            Type stateMachineContract) =>
+            RegisterStateMachine(builder, stateMachineImplementation, stateMachineContract);
 
+        [Obsolete("Use RegisterStateMachine(builder, implementation, contract, Lifetime.Singleton) instead.")]
         public static void AddSingletonStateMachine(
             this ContainerBuilder builder,
             Type stateMachineImplementation,
-            Type stateMachineContract)
-        {
-            AddStateMachineInternal(builder, stateMachineImplementation, stateMachineContract, Lifetime.Singleton);
-        }
+            Type stateMachineContract) =>
+            RegisterStateMachine(builder, stateMachineImplementation, stateMachineContract, Lifetime.Singleton);
 
-        public static void AddState(this ContainerBuilder builder, Type state)
-        {
-            ValidateStateBindingInput(state);
+        [Obsolete("Use RegisterState(builder, state) or the overload with Lifetime.")]
+        public static void AddState(this ContainerBuilder builder, Type state) =>
+            RegisterState(builder, state);
 
-            builder.AddTransient(state, GetStateContracts(state));
-        }
+        [Obsolete("Use RegisterState(builder, implementation, contract) or the overload with Lifetime.")]
+        public static void AddState(this ContainerBuilder builder, Type stateImplementation, Type stateContract) =>
+            RegisterState(builder, stateImplementation, stateContract);
 
-        public static void AddState(this ContainerBuilder builder, Type stateImplementation, Type stateContract)
-        {
-            ValidateStateBindingInput(stateImplementation, stateContract);
+        [Obsolete("Use RegisterState(builder, state, Lifetime.Singleton) instead.")]
+        public static void AddSingletonState(this ContainerBuilder builder, Type state) =>
+            RegisterState(builder, state, Lifetime.Singleton);
 
-            builder.AddTransient(stateImplementation, stateContract);
-        }
-
-        public static void AddSingletonState(this ContainerBuilder builder, Type state)
-        {
-            ValidateStateBindingInput(state);
-
-            builder.AddSingleton(state, GetStateContracts(state));
-        }
-
-        public static void AddSingletonState(this ContainerBuilder builder, Type stateImplementation,
-            Type stateContract)
-        {
-            ValidateStateBindingInput(stateImplementation, stateContract);
-
-            builder.AddSingleton(stateImplementation, stateContract);
-        }
+        [Obsolete("Use RegisterState(builder, implementation, contract, Lifetime.Singleton) instead.")]
+        public static void AddSingletonState(
+            this ContainerBuilder builder,
+            Type stateImplementation,
+            Type stateContract) =>
+            RegisterState(builder, stateImplementation, stateContract, Lifetime.Singleton);
 
         private static void ValidateStateBindingInput(Type stateImplementation, Type stateContract)
         {
@@ -62,7 +97,7 @@ namespace UniState
             if (!stateContract.IsAssignableFrom(stateImplementation))
             {
                 throw new ArgumentException(
-                    $"AddState({stateImplementation.Name}): Type parameter state must implement {stateContract.Name}.");
+                    $"RegisterState({stateImplementation.Name}): Type parameter state must implement {stateContract.Name}.");
             }
         }
 
@@ -71,7 +106,7 @@ namespace UniState
             if (!typeof(IExecutableState).IsAssignableFrom(state))
             {
                 throw new ArgumentException(
-                    $"AddState({state.Name}): Type parameter state must implement IState<TPayload>");
+                    $"RegisterState({state.Name}): Type parameter state must implement IState<TPayload>");
             }
         }
 
@@ -94,26 +129,26 @@ namespace UniState
             if (stateMachineImplementation == stateMachineContract)
             {
                 throw new ArgumentException(
-                    $"AddStateMachine<{stateMachineImplementation.Name}>: Type parameters must differ : " +
-                    "use AddStateMachine() where stateMachineImplementation implements stateMachineContract.\");");
+                    $"RegisterStateMachine<{stateMachineImplementation.Name}>: Type parameters must differ : " +
+                    "use RegisterStateMachine() where stateMachineImplementation implements stateMachineContract.\");");
             }
 
             if (!stateMachineContract.IsAssignableFrom(stateMachineImplementation))
             {
                 throw new ArgumentException(
-                    $"AddStateMachine: Type {stateMachineImplementation.Name} " +
+                    $"RegisterStateMachine: Type {stateMachineImplementation.Name} " +
                     $"must implement {stateMachineContract.Name}.");
             }
 
             if (!typeof(IStateMachine).IsAssignableFrom(stateMachineContract))
             {
                 throw new ArgumentException(
-                    $"AddStateMachine: Type {stateMachineContract.Name} " +
+                    $"RegisterStateMachine: Type {stateMachineContract.Name} " +
                     $"must implement IStateMachine.");
             }
         }
 
-        private static void AddStateMachineInternal(
+        private static void RegisterStateMachineInternal(
             ContainerBuilder builder,
             Type stateMachineImplementation,
             Type stateMachineContract,
@@ -121,62 +156,17 @@ namespace UniState
         {
             ValidateStateMachineBindingInput(stateMachineImplementation, stateMachineContract);
 
-            builder.Bindings.Add(Binding.Validated(
-                new ReflexStateMachineResolver(stateMachineImplementation, lifetime),
+            builder.RegisterFactory(
+                container =>
+                {
+                    var stateMachine = (IStateMachine)container.Construct(stateMachineImplementation);
+                    stateMachine.SetResolver(container.ToTypeResolver());
+                    return stateMachine;
+                },
                 stateMachineImplementation,
-                stateMachineImplementation,
-                stateMachineContract));
-        }
-
-        private sealed class ReflexStateMachineResolver : IResolver
-        {
-            private readonly Type _stateMachineImplementation;
-            private readonly Lifetime _lifetime;
-            private readonly List<IDisposable> _disposables = new();
-
-            private object _instance;
-
-            public Lifetime Lifetime => _lifetime;
-
-            public ReflexStateMachineResolver(Type stateMachineImplementation, Lifetime lifetime)
-            {
-                _stateMachineImplementation = stateMachineImplementation;
-                _lifetime = lifetime;
-            }
-
-            public object Resolve(Container container)
-            {
-                if (_lifetime == Lifetime.Singleton && _instance != null)
-                {
-                    return _instance;
-                }
-
-                var stateMachine = (IStateMachine)container.Construct(_stateMachineImplementation);
-                stateMachine.SetResolver(container.ToTypeResolver());
-
-                if (_lifetime == Lifetime.Singleton)
-                {
-                    _instance = stateMachine;
-                }
-
-                if (stateMachine is IDisposable disposable)
-                {
-                    _disposables.Add(disposable);
-                }
-
-                return stateMachine;
-            }
-
-            public void Dispose()
-            {
-                for (var i = _disposables.Count - 1; i >= 0; i--)
-                {
-                    _disposables[i].Dispose();
-                }
-
-                _disposables.Clear();
-                _instance = null;
-            }
+                new[] { stateMachineImplementation, stateMachineContract },
+                lifetime,
+                Resolution.Lazy);
         }
     }
 }

--- a/Assets/UniStateTests/Common/TestFixture/ReflexTestsBase.cs
+++ b/Assets/UniStateTests/Common/TestFixture/ReflexTestsBase.cs
@@ -1,5 +1,6 @@
 using Cysharp.Threading.Tasks;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 
 namespace UniStateTests.Common
@@ -37,7 +38,7 @@ namespace UniStateTests.Common
 
         protected virtual void SetupBindings(ContainerBuilder builder)
         {
-            builder.AddSingleton(typeof(ExecutionLogger));
+            builder.RegisterType(typeof(ExecutionLogger), Lifetime.Singleton, Resolution.Lazy);
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/BehaviorAttributeTests/BehaviorAttributeReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/BehaviorAttributeTests/BehaviorAttributeReflexTests.cs
@@ -2,6 +2,7 @@ using System.Collections;
 using Cysharp.Threading.Tasks;
 using NUnit.Framework;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 using UniStateTests.Common;
 using UniStateTests.PlayMode.StateBehaviorAttributeTests.Infrastructure;
@@ -22,11 +23,11 @@ namespace UniStateTests.PlayMode.StateBehaviorAttributeTests
         {
             base.SetupBindings(builder);
 
-            builder.AddStateMachine(typeof(StateMachineBehaviourAttribute), typeof(IVerifiableStateMachine));
-            builder.AddState(typeof(FirstState));
-            builder.AddState(typeof(NoReturnState));
-            builder.AddState(typeof(FastInitializeState));
-            builder.AddSingleton(typeof(BehaviourAttributeTestHelper));
+            builder.RegisterStateMachine(typeof(StateMachineBehaviourAttribute), typeof(IVerifiableStateMachine));
+            builder.RegisterState(typeof(FirstState));
+            builder.RegisterState(typeof(NoReturnState));
+            builder.RegisterState(typeof(FastInitializeState));
+            builder.RegisterType(typeof(BehaviourAttributeTestHelper), Lifetime.Singleton, Resolution.Lazy);
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/ExecutionTests/ExecutionReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/ExecutionTests/ExecutionReflexTests.cs
@@ -2,6 +2,7 @@ using System.Collections;
 using Cysharp.Threading.Tasks;
 using NUnit.Framework;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 using UniStateTests.Common;
 using UniStateTests.PlayMode.Execution.Infrastructure;
@@ -38,12 +39,12 @@ namespace UniStateTests.PlayMode.Execution
         {
             base.SetupBindings(builder);
 
-            builder.AddSingleton(typeof(ExecutionTestHelper));
-            builder.AddStateMachine(typeof(ExecutionStateMachine), typeof(IVerifiableStateMachine));
-            builder.AddState(typeof(FirstState));
-            builder.AddState(typeof(SecondState));
-            builder.AddState(typeof(SecondStateWithException));
-            builder.AddState(typeof(SecondStateWithWrongDependency));
+            builder.RegisterType(typeof(ExecutionTestHelper), Lifetime.Singleton, Resolution.Lazy);
+            builder.RegisterStateMachine(typeof(ExecutionStateMachine), typeof(IVerifiableStateMachine));
+            builder.RegisterState(typeof(FirstState));
+            builder.RegisterState(typeof(SecondState));
+            builder.RegisterState(typeof(SecondStateWithException));
+            builder.RegisterState(typeof(SecondStateWithWrongDependency));
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/GoBackTests/GoBackReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/GoBackTests/GoBackReflexTests.cs
@@ -2,6 +2,7 @@ using System.Collections;
 using Cysharp.Threading.Tasks;
 using NUnit.Framework;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 using UniStateTests.Common;
 using UniStateTests.PlayMode.GoBackTests.Infrastructure;
@@ -23,11 +24,11 @@ namespace UniStateTests.PlayMode.GoBackTests
         {
             base.SetupBindings(builder);
 
-            builder.AddSingleton(typeof(GoBackTestHelper));
-            builder.AddStateMachine(typeof(StateMachineGoBack), typeof(IVerifiableStateMachine));
-            builder.AddState(typeof(StateGoBackFirst));
-            builder.AddState(typeof(StateGoBackSecond));
-            builder.AddState(typeof(StateGoBackThird));
+            builder.RegisterType(typeof(GoBackTestHelper), Lifetime.Singleton, Resolution.Lazy);
+            builder.RegisterStateMachine(typeof(StateMachineGoBack), typeof(IVerifiableStateMachine));
+            builder.RegisterState(typeof(StateGoBackFirst));
+            builder.RegisterState(typeof(StateGoBackSecond));
+            builder.RegisterState(typeof(StateGoBackThird));
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/GoBackToTests/GoBackToReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/GoBackToTests/GoBackToReflexTests.cs
@@ -1,5 +1,6 @@
 using NUnit.Framework;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 using UniStateTests.Common;
 using UniStateTests.PlayMode.GoBackToTests.Infrastructure;
@@ -17,12 +18,12 @@ namespace UniStateTests.PlayMode.GoBackToTests
         {
             base.SetupBindings(builder);
 
-            builder.AddStateMachine(typeof(GoBackToStateMachine), typeof(IVerifiableStateMachine));
-            builder.AddState(typeof(GoBackToState1));
-            builder.AddState(typeof(GoBackToState2));
-            builder.AddState(typeof(GoBackToState3));
-            builder.AddState(typeof(GoBackToState4));
-            builder.AddSingleton(typeof(GoBackToTestsHelper));
+            builder.RegisterStateMachine(typeof(GoBackToStateMachine), typeof(IVerifiableStateMachine));
+            builder.RegisterState(typeof(GoBackToState1));
+            builder.RegisterState(typeof(GoBackToState2));
+            builder.RegisterState(typeof(GoBackToState3));
+            builder.RegisterState(typeof(GoBackToState4));
+            builder.RegisterType(typeof(GoBackToTestsHelper), Lifetime.Singleton, Resolution.Lazy);
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/GoToStateTests/GoToReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/GoToStateTests/GoToReflexTests.cs
@@ -22,20 +22,20 @@ namespace UniStateTests.PlayMode.GoToStateTests
         {
             base.SetupBindings(builder);
 
-            builder.AddStateMachine(typeof(StateMachineGoToState), typeof(IVerifiableStateMachine));
-            builder.AddState(typeof(StateGoTo1));
-            builder.AddState(typeof(StateGoTo2));
-            builder.AddState(typeof(StateGoTo3));
-            builder.AddState(typeof(StateGoTo3), typeof(StateGoToAbstract3));
-            builder.AddState(typeof(StateGoTo4));
-            builder.AddState(typeof(StateGoTo5));
-            builder.AddState(typeof(CompositeStateGoTo6));
-            builder.AddState(typeof(SubStateGoTo6First));
-            builder.AddState(typeof(SubStateGoTo6Second));
-            builder.AddState(typeof(CompositeStateGoTo7));
-            builder.AddState(typeof(SubStateGoTo7First));
-            builder.AddState(typeof(SubStateGoTo7Second));
-            builder.AddState(typeof(StateGoTo8));
+            builder.RegisterStateMachine(typeof(StateMachineGoToState), typeof(IVerifiableStateMachine));
+            builder.RegisterState(typeof(StateGoTo1));
+            builder.RegisterState(typeof(StateGoTo2));
+            builder.RegisterState(typeof(StateGoTo3));
+            builder.RegisterState(typeof(StateGoTo3), typeof(StateGoToAbstract3));
+            builder.RegisterState(typeof(StateGoTo4));
+            builder.RegisterState(typeof(StateGoTo5));
+            builder.RegisterState(typeof(CompositeStateGoTo6));
+            builder.RegisterState(typeof(SubStateGoTo6First));
+            builder.RegisterState(typeof(SubStateGoTo6Second));
+            builder.RegisterState(typeof(CompositeStateGoTo7));
+            builder.RegisterState(typeof(SubStateGoTo7First));
+            builder.RegisterState(typeof(SubStateGoTo7Second));
+            builder.RegisterState(typeof(StateGoTo8));
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/HistoryTests/HistorySizeReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/HistoryTests/HistorySizeReflexTests.cs
@@ -2,6 +2,7 @@ using System.Collections;
 using Cysharp.Threading.Tasks;
 using NUnit.Framework;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 using UniStateTests.Common;
 using UniStateTests.PlayMode.HistoryTests.Infrastructure;
@@ -24,13 +25,13 @@ namespace UniStateTests.PlayMode.HistoryTests
         {
             base.SetupBindings(builder);
 
-            builder.AddSingleton(typeof(HistorySizeTestHelper));
-            builder.AddStateMachine(typeof(StateMachineLongHistory), typeof(IStateMachineLongHistory));
-            builder.AddStateMachine(typeof(StateMachineZeroHistory), typeof(IStateMachineZeroHistory));
-            builder.AddState(typeof(StateInitLongHistory));
-            builder.AddState(typeof(StateInitZeroHistory));
-            builder.AddState(typeof(StateFooHistory));
-            builder.AddState(typeof(StateBarHistory));
+            builder.RegisterType(typeof(HistorySizeTestHelper), Lifetime.Singleton, Resolution.Lazy);
+            builder.RegisterStateMachine(typeof(StateMachineLongHistory), typeof(IStateMachineLongHistory));
+            builder.RegisterStateMachine(typeof(StateMachineZeroHistory), typeof(IStateMachineZeroHistory));
+            builder.RegisterState(typeof(StateInitLongHistory));
+            builder.RegisterState(typeof(StateInitZeroHistory));
+            builder.RegisterState(typeof(StateFooHistory));
+            builder.RegisterState(typeof(StateBarHistory));
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/RecoveryTransitionTests/RecoveryReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/RecoveryTransitionTests/RecoveryReflexTests.cs
@@ -2,6 +2,7 @@ using System.Collections;
 using Cysharp.Threading.Tasks;
 using NUnit.Framework;
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 using UniStateTests.Common;
 using UniStateTests.PlayMode.RecoveryTransitionTests.Infrastructure;
@@ -28,14 +29,14 @@ namespace UniStateTests.PlayMode.RecoveryTransitionTests
         {
             base.SetupBindings(builder);
 
-            builder.AddSingleton(typeof(RecoveryTestHelper));
-            builder.AddStateMachine(typeof(StateMachineDefaultRecovery), typeof(IStateMachineDefaultRecovery));
-            builder.AddStateMachine(typeof(StateMachineGoToStateRecovery), typeof(IStateMachineGoToStateRecovery));
-            builder.AddStateMachine(typeof(StateMachineExitRecovery), typeof(IStateMachineExitRecovery));
-            builder.AddState(typeof(StateInitial));
-            builder.AddState(typeof(StateThrowTwoException));
-            builder.AddState(typeof(StateWithFailExecution));
-            builder.AddState(typeof(StateStartedAfterException));
+            builder.RegisterType(typeof(RecoveryTestHelper), Lifetime.Singleton, Resolution.Lazy);
+            builder.RegisterStateMachine(typeof(StateMachineDefaultRecovery), typeof(IStateMachineDefaultRecovery));
+            builder.RegisterStateMachine(typeof(StateMachineGoToStateRecovery), typeof(IStateMachineGoToStateRecovery));
+            builder.RegisterStateMachine(typeof(StateMachineExitRecovery), typeof(IStateMachineExitRecovery));
+            builder.RegisterState(typeof(StateInitial));
+            builder.RegisterState(typeof(StateThrowTwoException));
+            builder.RegisterState(typeof(StateWithFailExecution));
+            builder.RegisterState(typeof(StateStartedAfterException));
         }
     }
 }

--- a/Assets/UniStateTests/PlayMode/SubStateTests/SubStateReflexTests.cs
+++ b/Assets/UniStateTests/PlayMode/SubStateTests/SubStateReflexTests.cs
@@ -20,13 +20,13 @@ namespace UniStateTests.PlayMode.SubStateTests
         {
             base.SetupBindings(builder);
 
-            builder.AddStateMachine(typeof(StateMachineSubStates), typeof(IVerifiableStateMachine));
-            builder.AddState(typeof(StateInitial));
-            builder.AddState(typeof(StateFinal));
-            builder.AddState(typeof(SubStateInitialFirst));
-            builder.AddState(typeof(SubStateInitialSecond));
-            builder.AddState(typeof(SubStateFinalFirst));
-            builder.AddState(typeof(SubStateFinalSecond));
+            builder.RegisterStateMachine(typeof(StateMachineSubStates), typeof(IVerifiableStateMachine));
+            builder.RegisterState(typeof(StateInitial));
+            builder.RegisterState(typeof(StateFinal));
+            builder.RegisterState(typeof(SubStateInitialFirst));
+            builder.RegisterState(typeof(SubStateInitialSecond));
+            builder.RegisterState(typeof(SubStateFinalFirst));
+            builder.RegisterState(typeof(SubStateFinalSecond));
         }
     }
 }

--- a/Packages/manifest.json
+++ b/Packages/manifest.json
@@ -1,7 +1,7 @@
 {
   "dependencies": {
     "com.cysharp.unitask": "https://github.com/Cysharp/UniTask.git?path=src/UniTask/Assets/Plugins/UniTask",
-    "com.gustavopsantos.reflex": "https://github.com/gustavopsantos/reflex.git?path=/Assets/Reflex/#11.0.0",
+    "com.gustavopsantos.reflex": "https://github.com/gustavopsantos/reflex.git?path=/Assets/Reflex/#14.3.0",
     "com.svermeulen.extenject": "https://github.com/Mathijs-Bakker/Extenject.git?path=UnityProject/Assets/Plugins/Zenject/Source",
     "com.unity.ide.rider": "3.0.36",
     "com.unity.render-pipelines.universal": "12.1.7",

--- a/Packages/packages-lock.json
+++ b/Packages/packages-lock.json
@@ -8,11 +8,11 @@
       "hash": "809d23edae0df58fd0f61b4b6f67855c59d1b2fb"
     },
     "com.gustavopsantos.reflex": {
-      "version": "https://github.com/gustavopsantos/reflex.git?path=/Assets/Reflex/#11.0.0",
+      "version": "https://github.com/gustavopsantos/reflex.git?path=/Assets/Reflex/#14.3.0",
       "depth": 0,
       "source": "git",
       "dependencies": {},
-      "hash": "224d2a1d1716b26f8eddf2e0d4a5c47d50d7a75c"
+      "hash": "86569ca217b989a8cec9db1e37f81c44130d23f1"
     },
     "com.svermeulen.extenject": {
       "version": "https://github.com/Mathijs-Bakker/Extenject.git?path=UnityProject/Assets/Plugins/Zenject/Source",

--- a/README.md
+++ b/README.md
@@ -1189,28 +1189,30 @@ namespace Examples.Infrastructure.Reflex
 All state machines, states, and their dependencies should be registered in the DI container using Reflex's
 `ContainerBuilder`. Special extension methods have been provided for convenient registration.
 
-`AddState(typeof(MyState))` and `AddSingletonState(typeof(MyState))` register the state as itself and as all implemented
-interfaces. Use the two-parameter overload when you need to expose an additional abstract/base contract explicitly.
+`RegisterState(typeof(MyState))` registers the state as itself and as all implemented interfaces.
+Use the overload with `Lifetime.Singleton` when you need a singleton lifetime.
+Use the two-parameter overload when you need to expose an additional abstract/base contract explicitly.
 
 Here's example code demonstrating the available extension methods:
 
 ```csharp
 using Reflex.Core;
+using Reflex.Enums;
 using UniState;
 
 private void RegisterStates(ContainerBuilder builder)
 {
     // Recommended usage for general cases
 
-    builder.AddStateMachine(typeof(StateMachine), typeof(IStateMachine));
-    builder.AddState(typeof(BarState));
-    builder.AddState(typeof(BarState), typeof(BarBaseState));
+    builder.RegisterStateMachine(typeof(StateMachine), typeof(IStateMachine));
+    builder.RegisterState(typeof(BarState));
+    builder.RegisterState(typeof(BarState), typeof(BarBaseState));
     
     // Singleton version (use cautiously, not recommended in most cases)
 
-    builder.AddSingletonStateMachine(typeof(StateMachine), typeof(IStateMachine));
-    builder.AddSingletonState(typeof(BarState));
-    builder.AddSingletonState(typeof(BarState), typeof(BarBaseState));
+    builder.RegisterStateMachine(typeof(StateMachine), typeof(IStateMachine), Lifetime.Singleton);
+    builder.RegisterState(typeof(BarState), Lifetime.Singleton);
+    builder.RegisterState(typeof(BarState), typeof(BarBaseState), Lifetime.Singleton);
 }
 ```
 

